sympy/solvers/ode/tests/test_systems.py

```diff
@@ -1273,7 +1269,32 @@ def test_sysode_linear_neq_order1_type2():
     assert dsolve(eq26) == sol26
     assert checksysodesol(eq26 , sol26) == (True , [0,0])
 
+    # Test Case added for issue #22715
+    # https://github.com/sympy/sympy/issues/22715
 
+    import sympy as sym
+
+    def eqnsubs(eqns, vartosub, valtosub):
+        subeqn = eqns.copy()
+        sublist = list(zip(vartosub, valtosub))
+        for idx, eqn in enumerate(subeqn):
+            subeqn[idx] = subeqn[idx].subs(sublist)
+        return subeqn
+
+    sym.var('a b c d e f t')
+    sym.var('x y', cls=sym.Function)
+
+    eqn1 = sym.Eq(sym.diff(x(t),t),a*x(t)+b*y(t)+c)
+    eqn2 = sym.Eq(sym.diff(y(t),t),d*x(t)+e*y(t)+f)
+    eqns = [eqn1, eqn2]
+
+    eq27 = eqnsubs(eqns, (a, b, c, d, e, f), (0, -1, 10, 5, -2, 3))
```

Comment:
It would be better to write this like all the other examples in the tests. We don't need the `eqnsubs` function. Just define the equations directly.